services/dashboard/app/components/Drawer.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { ReactNode } from "react";
+import { createPortal } from "react-dom";
+
+interface DrawerProps {
+  /** Controls visibility of the drawer */
+  open: boolean;
+  /** Callback fired when the user clicks on the overlay or the close button */
+  onClose: () => void;
+  /** Optional fixed width utility class (defaults to w-96) */
+  widthClass?: string;
+  /** Optional ARIA label/title for the drawer */
+  title?: string;
+  /** Drawer body */
+  children: ReactNode;
+}
+
+/**
+ * Basic right-anchored slide-over panel.
+ *
+ * Renders through a React portal so it is mounted at the document <body> level and
+ * therefore not affected by parent stacking-contexts.
+ */
+export function Drawer({
+  open,
+  onClose,
+  widthClass = "w-96",
+  title,
+  children,
+}: DrawerProps) {
+  if (!open || typeof document === "undefined") return null;
+
+  return createPortal(
+    <div className="fixed inset-0 z-50 flex">
+      {/* Overlay */}
+      <div
+        aria-label="Close drawer overlay"
+        className="flex-1 bg-black/50 "
```

Comment:
I haven't used black backgrounds? Can you use a neutral dark color? Also can you provide a dark theme alternative?